src/modules/catalog/database/product.in-memory.repository.ts

```diff
@@ -33,4 +34,10 @@ export class ProductInMemoryRepository
 
     return Promise.resolve(this.toDomainEntities(filteredProducts));
   }
+  async deleteArchive(product: ProductEntity): Promise<any> {
+    if (ProductStatus.DELETED === product.getPropsCopy().status) {
```

Comment:
please switch equation, to be read as status is equal deleted is more reasonable then the reverse